src/hooks/useCivicProfile.js

```diff
@@ -0,0 +1,99 @@
+import { useQuery, useMutation, useQueryClient } from '@tanstack/react-query';
+import {
+  getDate,
+  getInteger,
+  getStringNoLocale,
+  getThing,
+  buildThing,
+  createThing,
+  setThing,
+  createSolidDataset,
+  getSolidDataset,
+  saveSolidDatasetAt
+} from '@inrupt/solid-client';
+import { RDF_PREDICATES } from '@constants';
+import useSession from './useSession';
+
+const makeIntoThing = ({ firstName, lastName, dateOfBirth, gender }) =>
+  buildThing(createThing({ name: 'Civic Profile' }))
+    .addStringNoLocale(RDF_PREDICATES.legalFirstName, firstName)
+    .addStringNoLocale(RDF_PREDICATES.legalLastName, lastName)
+    .addDate(RDF_PREDICATES.legalDOB, dateOfBirth)
+    .addInteger(RDF_PREDICATES.legalGender, gender)
+    .build();
+
+let storedDataset;
+const useCivicProfile = () => {
+  const queryClient = useQueryClient();
+  const { session, podUrl } = useSession();
+  const { fetch } = session;
+  const fileUrl = podUrl && new URL('PASS/AdditionalProfiles/civic_profile.ttl', podUrl).toString();
```

Comment:
One small thing is, we've already created a path previously for additional profile files at `PASS/Profile/` in theie pods. So was thinking we could do the same thing here.

Reply:
Sure